ee/src/main/java/org/jboss/as/ee/structure/ComponentAggregationProcessor.java

```diff
@@ -118,4 +123,29 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
             }
         }
     }
+
+    private static final class ValueService<T> implements Service<T> {
```

Comment:
@ropalka This class seems overly complicated, as 'componentRegistry' is a final var that cannot be null. I think this can just be a no-op start/stop and return componentRegistry from getValue.

Reply:
It was refactored out from JBoss MSC code base so I didn't optimize it further @bstansberry . It is fixed now.